Automattic/newspack-popups
# feat: reconcile different reader sessions

### All Submissions:






### Changes proposed in this Pull Request:



Handles the case of multiple reader sessions "belonging" to the same reader. As soon as the reader takes an action that provides a unique identifier (email address or WP user ID), we can look up any past client IDs that are associated with the same identifier. Data from all past sessions are reconciled with the current session and used to segment the reader.

Closes #901.

### How to test the changes in this Pull Request:

1. In a new incognito session, subscribe to a newsletter and view several articles. Note how many article and category views you log (using the `newspack-campaigns-debug` param). Keep this session open.
2. In a separate browser or device, visit the site and sign in using the same email address.
3. In the same session, navigate to a new page and in the debug response confirm that your article and category view counts from step one got carried over to the new session.
4. Also confirm that you're segmented as a subscriber in the new session.
5. Repeat with a few different sessions and confirm that each time, your data and reader events from all sessions are carried over to the new session. Maybe make a donation in one of these sessions.
6. Go back to the first session and navigate to a new page. Confirm that your activity from other sessions (including the donation) is carried back to the first session.

### Other information:
